utils.c

```diff
@@ -18,10 +18,22 @@
 #include <dirent.h>
 #include <errno.h>
 #include <err.h>
+#include <unistd.h>
 
 #include "version.h"
 #include "utils.h"
 
+void restrict_fsaccess(const char **rootdir)
+{
+	int chrooted = !chroot(*rootdir);
+	assert(chrooted || errno == EPERM);
```

Comment:
Please issue an error message for failures of `chroot(2)`. The bare minimum is to report `errno` and `strerror(errno)`, but you can have a `switch..case` if you can improve over the messages of `strerror(3)`.

The relevance of reporting errors from `chroot()` are errors such as `EACCES`, `ELOOP`, and `ENOENT`.